### PR TITLE
CloudEvents implementation for Kafka messages

### PR DESCRIPTION
**Pull request by sansar715**
CloudEvents implementation - Sender and Receiver for CloudEvents over kafka

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -28,25 +36,35 @@ github.com/go-redis/redis v6.15.2+incompatible/go.mod h1:NAIEuMOZ/fxfXJIrKDQDz8w
 github.com/golang/protobuf v1.2.0/go.mod h1:6lQm79b+lXiMfvg/cZm0SGofjICqVBUtrP5yJMmIC1U=
 github.com/golang/snappy v0.0.1 h1:Qgr9rKW7uDUkrbSmQeiDsGa8SjGyCOGtuasMWwvp2P4=
 github.com/golang/snappy v0.0.1/go.mod h1:/XxbfmMg8lxefKM7IXC3fBNl/7bRcc72aCRzEWrmP2Q=
+github.com/google/go-cmp v0.3.0/go.mod h1:8QqcDgzrUqlUb/G2PQTWiueGozuR1884gddMywk6iLU=
```
Comment (rosasck) [inline go.sum:39]: run `go mod tidy`

File: stream/kakfa.go [hunk 2]
```diff
@@ -151,5 +158,90 @@ func NewKafkaStream(config KafkaStreamConfig) (Stream, error) {
...
+	if err != nil {
+		log.Fatalf("Failed to create receiver: %s", err.Error())
+	}
+	kafkaStream.receiver = receiver
+
 	return kafkaStream, nil
 }
+
+func (ks *KafkaStream) Send(event CloudEvent) error {
+	//defer ks.sender.Close(context.Background())
```
Comment (rosasck) [inline stream/kakfa.go:178]: do we need to close the channel after a send?

File: stream/kakfa.go [hunk 3]
```diff
@@ -151,5 +158,90 @@ func NewKafkaStream(config KafkaStreamConfig) (Stream, error) {
...
+		cloudEvent,
+	); cloudevents.IsUndelivered(result) {
+		log.Printf("Failed to send: %v", result)
+	} else {
+		log.Printf("Message accepted: %t", cloudevents.IsACK(result))
+	}
+	return nil
+}
+
+func (ks *KafkaStream) Receive(close chan struct{}) (<-chan CloudEvent, error) {
```
Comment (rosasck) [inline stream/kakfa.go:197]: Add Function comments

File: stream/stream.go [hunk 4]
```diff
@@ -59,3 +76,9 @@ func (c *NoOpEventClient) PublishData(tag string, data auth.Claims) error {
 	// do nothing! event gets dropped.
 	return nil
 }
+
+func (c *NoOpEventClient) PublishCloudEvent(tag string, eventType string, eventSource string, contentType string,
```
Comment (rosasck) [inline stream/stream.go:80]: comment

**Review by rosasck — APPROVED** [review]

**Comment by sansar715** [comment]
created a different PR - https://github.com/tozny/utils-go/pull/62 - for this since the commits needed to be signed

**Pull request closed without merging**